packages/pymedphys_deliverydata/src/pymedphys_deliverydata/dicom/to_dicom.py

```diff
@@ -0,0 +1,160 @@
+from pymedphys_base.deliverydata import DeliveryDataBase
+
+from pymedphys_dicom.rtplan import (
+    get_gantry_angles_from_dicom,
+    build_control_points,
+    replace_fraction_group,
+    replace_beam_sequence,
+    restore_trailing_zeros,
+    merge_beam_sequences,
+    get_fraction_group_index,
+    convert_to_one_fraction_group)
+
+from ..utilities import get_all_masked_delivery_data
+
+
+def delivery_data_to_dicom(filtered_delivery_data: DeliveryDataBase,
+                           dicom_template,
+                           fraction_group_number):
+    single_fraction_group_template = convert_to_one_fraction_group(
+        dicom_template, fraction_group_number)
+
+    template_gantry_angles = get_gantry_angles_from_dicom(
+        single_fraction_group_template)
+
+    gantry_tol = gantry_tol_from_gantry_angles(template_gantry_angles)
+
+    all_masked_delivery_data = get_all_masked_delivery_data(
+        filtered_delivery_data, template_gantry_angles, gantry_tol)
+
+    fraction_group_index = get_fraction_group_index(
+        single_fraction_group_template, fraction_group_number)
+
+    single_beam_dicoms = []
+    for beam_index, masked_delivery_data in enumerate(all_masked_delivery_data):
+        single_beam_dicoms.append(delivery_data_to_dicom_single_beam(
+            masked_delivery_data, single_fraction_group_template, beam_index,
+            fraction_group_index))
+
+    return merge_beam_sequences(single_beam_dicoms)
+
+
+def delivery_data_to_dicom_single_beam(delivery_data, dicom_template,
+                                       beam_index, fraction_group_index):
+
+    created_dicom = deepcopy(dicom_template)
+    data_converted = coordinate_convert_delivery_data(delivery_data)
+
+    beam = created_dicom.BeamSequence[beam_index]
+    cp_sequence = beam.ControlPointSequence
+    initial_cp = cp_sequence[0]
+    subsequent_cp = cp_sequence[-1]
+
+    all_control_points = build_control_points(
+        initial_cp, subsequent_cp, data_converted)
+
+    beam_meterset = '{0:.6f}'.format(data_converted['monitor_units'][-1])
+    replace_fraction_group(
+        created_dicom, beam_meterset, beam_index, fraction_group_index)
+    replace_beam_sequence(created_dicom, all_control_points, beam_index)
+
+    restore_trailing_zeros(created_dicom)
+
+    return created_dicom
+
+
+def coordinate_convert_delivery_data(delivery_data):
+    monitor_units = delivery_data.monitor_units
+    mlc = mlc_dd2dcm(delivery_data.mlc)
+    jaw = jaw_dd2dcm(delivery_data.jaw)
+    gantry_angle, gantry_movement = angle_dd2dcm(delivery_data.gantry)
+    collimator_angle, collimator_movement = angle_dd2dcm(
+        delivery_data.collimator)
+
+    return {
+        'monitor_units': monitor_units,
+        'mlc': mlc,
+        'jaw': jaw,
+        'gantry_angle': gantry_angle,
+        'gantry_movement': gantry_movement,
+        'collimator_angle': collimator_angle,
+        'collimator_movement': collimator_movement
+    }
+
+
+def jaw_dd2dcm(jaw):
```

Comment:
`dd2dcm`? This is sooooo un-SimonBiggsy.

And you complain about my preference for `ds`???

(you don't actually complain, I'm playing)

Reply:
:)